PWGHF/TableProducer/treeCreatorDstarToD0Pi.cxx

```diff
@@ -0,0 +1,537 @@
+                  full::NSigTpcTofPi0,
+                  full::NSigTpcTofKa0,
+                  full::NSigTpcPi1,
+                  full::NSigTpcKa1,
+                  full::NSigTofPi1,
+                  full::NSigTofKa1,
+                  full::NSigTpcTofPi1,
+                  full::NSigTpcTofKa1,
+                  full::NSigTpcPiSoftPi,
+                  full::NSigTpcKaSoftPi,
+                  full::NSigTofPiSoftPi,
+                  full::NSigTofKaSoftPi,
+                  full::NSigTpcTofPiSoftPi,
+                  full::NSigTpcTofKaSoftPi,
+                  full::MD0,
+                  full::PtD0,
+                  full::PD0,
+                  full::CtD0,
+                  full::EtaD0,
+                  full::PhiD0,
+                  full::YD0,
+                  full::M,
+                  full::Pt,
+                  full::P,
+                  full::Eta,
+                  full::Phi,
+                  full::Y,
+                  full::E,
+                  full::CandidateSelFlag,
+                  hf_cand_dstar::FlagMcMatchRec,
+                  hf_cand_dstar::OriginMcRec);
+
+DECLARE_SOA_TABLE(HfCandDstFullEvs, "AOD", "HFCANDDSTFULLEV",
+                  collision::BCId,
+                  collision::NumContrib,
+                  collision::PosX,
+                  collision::PosY,
+                  collision::PosZ,
+                  full::IsEventReject,
+                  full::RunNumber);
+
+DECLARE_SOA_TABLE(HfCandDstFullPs, "AOD", "HFCANDDSTFULLP",
+                  collision::BCId,
+                  full::Pt,
+                  full::Eta,
+                  full::Phi,
+                  full::Y,
+                  hf_cand_dstar::FlagMcMatchGen,
+                  hf_cand_dstar::OriginMcGen);
+
+} // namespace o2::aod
+
+/// Writes the full information in an output TTree
+struct HfTreeCreatorDstarToD0Pi {
+  Produces<o2::aod::HfCandDstFulls> rowCandidateFull;
+  Produces<o2::aod::HfCandDstFullEvs> rowCandidateFullEvents;
+  Produces<o2::aod::HfCandDstFullPs> rowCandidateFullParticles;
+  Produces<o2::aod::HfCandDstLites> rowCandidateLite;
+
+  Configurable<bool> selectionFlagDstarToD0Pi{"selectionFlagDstarToD0Pi", true, "Selection Flag for D* decay to D0 & Pi"};
+  Configurable<bool> fillCandidateLiteTable{"fillCandidateLiteTable", false, "Switch to fill lite table with candidate properties"};
+  // parameters for production of training samples
+  Configurable<bool> fillOnlySignal{"fillOnlySignal", false, "Flag to fill derived tables with signal for ML trainings"};
+  Configurable<bool> fillOnlyBackground{"fillOnlyBackground", false, "Flag to fill derived tables with background for ML trainings"};
+  Configurable<float> downSampleBkgFactor{"downSampleBkgFactor", 1., "Fraction of background candidates to keep for ML trainings"};
+  Configurable<float> ptMaxForDownSample{"ptMaxForDownSample", 10., "Maximum pt for the application of the downsampling factor"};
+
+  HfHelper hfHelper;
+
```

Comment:
```suggestion
```

Reply:
I hope the compiler would complain about unused variable but just in case.

Reply:
Indeed, strange that it did not complain locally (maybe because it's a data member?)